guides/v2.1/howdoi/checkout/checkout-add-custom-carrier.md

```diff
@@ -0,0 +1,298 @@
+---
+layout: tutorial
+group: how-do-i
+subgroup:
+title: Add custom shipping carrier
+subtitle: Customize Checkout
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+menu_order: 5
+level3_subgroup: checkout-tutorial
+functional_areas:
+  - Checkout
+---
+
+This topic describes how to add a custom shipping carrier.
+
+To add a new shipping carrier to the Magento checkout:
+
+1. [Create a new module](#create-module)
+2. [Add the carrier configuration](#create-configuration)
+3. [Create the carrier model](#carrier-model)
+4. [Enable the module](#enable-module)
+
+## Step 1: Create a new module {#create-module}
+
+The example module for use here is `Vendor_CustomShipping`.
+
+### Source code of `app/code/Vendor/CustomShipping/registration.php`
+
+```php
+<?php
+
+use Magento\Framework\Component\ComponentRegistrar;
+
+ComponentRegistrar::register(
+    ComponentRegistrar::MODULE,
+    'Vendor_CustomShipping',
+    __DIR__
+);
+
+```
+
+### Source code of `app/code/Vendor/CustomShipping/composer.json`
+
+```json
+{
+    "name": "vendor/custom-shipping",
+    "description": "Custom shipping module",
+    "require": {
+        "php": "~5.6.5|7.0.2|7.0.4|~7.0.6|~7.1.0",
+        "magento/module-config": "100.1.*",
+        "magento/module-store": "100.1.*",
+        "magento/module-backend": "100.1.*",
+        "magento/module-shipping": "100.1.*",
+        "magento/module-catalog": "101.0.*",
+        "magento/module-sales-rule": "100.1.*",
+        "magento/module-directory": "100.1.*",
+        "magento/module-quote": "100.1.*",
+        "magento/framework": "100.1.*"
+    },
+    "type": "magento2-module",
+    "license": [
+        "OSL-3.0",
+        "AFL-3.0"
+    ],
+    "autoload": {
+        "files": [
+            "registration.php"
+        ],
+        "psr-4": {
+            "Vendor\\CustomShipping\\": ""
+        }
+    },
+    "version": "1.0.0"
+}
+```
+
+### Source code of `app/code/Vendor/CustomShipping/etc/module.xml`
+
+```xml
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Vendor_CustomShipping" >
+        <sequence>
+            <module name="Magento_Store"/>
+            <module name="Magento_Sales"/>
+            <module name="Magento_Quote"/>
+            <module name="Magento_SalesRule"/>
+        </sequence>
+    </module>
+</config>
+```
+
+## Step 2: Add module configuration {#create-configuration}
+
```

Comment:
We do not stack headings with no sentences in between. Add a sentence for this ##; Something like "To add a module configuration use the following source code snippets..."

Reply:
> We do not stack headings with no sentences in between. Add a sentence for this ##; Something like "To add a module configuration use the following source code snippets..."

**added**